lib/spack/spack/package.py

```diff
@@ -1420,6 +1524,43 @@ def rpath_args(self):
         return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
 
 
+class Package(PackageBase):
+    phases = ['install', 'log']
+    # This will be used as a registration decorator in user
+    # packages, if need be
+    PackageBase.sanity_check('install')(PackageBase.sanity_check_prefix)
+
+
+class AutotoolsPackage(PackageBase):
+    phases = ['autoreconf', 'configure', 'build', 'install', 'log']
```

Comment:
log == provenance? I think it's the part that copies build.out, build.env, spec.yaml, etc to prefix/.spack.